lib/manager/flux/artifacts.ts

```diff
@@ -0,0 +1,50 @@
+import { logger } from '../../logger';
+import { exec } from '../../util/exec';
+import type { ExecOptions } from '../../util/exec/types';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+import { isSystemManifest } from '.';
+
+export async function updateArtifacts({
+  packageFileName,
+  updatedDeps,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  if (!isSystemManifest(packageFileName) || !updatedDeps[0]?.newVersion) {
```

Comment:
Can you explain this requirement/limitation that "first updated dep must have a new version"? E.g. is it your intent that the system manifest is upgraded on its own and not mixed with other upgrades of dependencies?

Reply:
This logic is based on how system manifests are treated in `extract.ts`. Essentially, we treat a system manifest file as a single unit that has only one `github-releases` dependency on Flux itself. In other words, if `isSystemManifest(packageFileName)` returns `true` here, then `updatedDeps.length` will be `1`. This `updatedDeps[0]?.newVersion` check is only here as a defensive measure and should never actually fail.